service/src/test/java/bio/terra/workspace/service/iam/SamServiceTest.java

```diff
@@ -309,18 +306,16 @@ void workspaceReaderIsNotPrivateResourceReader() throws Exception {
 
   @Test
   void duplicateResourceCreateIgnored() throws Exception {
-    UUID workspaceId = createWorkspaceDefaultUser();
-
     ControlledResource bucketResource = defaultBucket(workspaceId).build();
     samService.createControlledResource(bucketResource, null, null, defaultUserRequest());
     // This duplicate call should complete without throwing.
     samService.createControlledResource(bucketResource, null, null, defaultUserRequest());
+    // Delete the bucket so we can clean up the workspace.
```

Comment:
Can't you just delete the workspace and let it clean up the bucket and other resources?

Reply:
Normally yes, but this test is a little weird because it's only creating the controlled resource in Sam, not WSM. I think it's helpful to verify this property of `SamService` here, but we do need to explicitly clean up the resource we create.